src/createQueryGroups.ts

```diff
@@ -54,8 +54,18 @@ export function createQueryGroups(options?: QueryGroupOptions) {
 
     actionGroups.get(actionKey)!.add(groupKey)
 
+    const onDispose = () => {
+      if (!groups.get(groupKey)?.active) {
+        groups.delete(groupKey)
+      }
+
+      options?.onDispose?.()
+    }
+
     if(!groups.has(groupKey)) {
-      groups.set(groupKey, createQueryGroup(action, parameters, options))
+      const group = createQueryGroup(action, parameters, {...options, onDispose})
+
+      groups.set(groupKey, group)
```

Comment:
I don't think this is quite right. Basically the groups onDispose ends up wrapping whatever the first query made to the group. Which means that queries callback is going to continue getting called even after its been disposed up.